hoomd/device.py

```diff
@@ -27,6 +27,42 @@
 from hoomd import _hoomd
 
 
+class NoticeFile:
+    """Write a notice message to a file using `hoomd.Device`.
+
+    Args:
+        device (`hoomd.Device`): The `Device` object.
+        level (int): Message notice level. Default value is 1
+    """
+
+    def __init__(self, device, level=1):
+        self._msg = device._cpp_msg
+        self._buff = ""
+        self._level = level
+
+    def write(self, message):
+        """Writes the notice message to the device.
+
+        Args:
+            message (str): Message to write.
+        """
+        self._buff += str(message)
+        lines = self._buff.split("\n")
+        for line in lines:
+
+            if "\n" == line[-2:]:
+                pass
+            else:
+                line += "\n"
+
+            self._msg.notice(self._level, line)
+        self._buff = ""
```

Comment:
This is not working as intended.
With this test script:
```
import hoomd
cpu = hoomd.device.CPU()
notice_file = hoomd.device.NoticeFile(cpu)

notice_file.write('t')
notice_file.write('e')
notice_file.write('s')
notice_file.write('t')
notice_file.write('1')
notice_file.write('\n')

notice_file.write('test2')
notice_file.write('\n')

notice_file.write('test3\ntest4\n')
```
I get the output:
```
t
e
s
t
1


test2


test3
test4

```
where I would expect:
```
test1
test2
test3
test4
```
(you can set `notice_file` to `sys.stdout` to confirm).

This change allows buffering of input missing `\n` over to the next call to `write` and avoids adding extra newlines to the output.
```suggestion
        lines = self._buff.split("\n")
        for line in lines[:-1]:
            self._msg.notice(self._level, line + "\n")

        if self._buff.endswith("\n"):
            self._msg_notice(self._level, lines[-1] + "\n")
            self._buff = ""
        else:
            self._buff = lines[-1]
```